sisl/io/orca/output.py

```diff
@@ -0,0 +1,400 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+import numpy as np
+from .sile import SileORCA
+from ..sile import add_sile, sile_fh_open
+
+from sisl.utils import PropertyDict
+from sisl._internal import set_module
+from sisl.unit import units
+
+__all__ = ['outputSileORCA']
+
+
+@set_module("sisl.io.orca")
+class outputSileORCA(SileORCA):
+    """ Output file from ORCA """
+
+    def _setup(self, *args, **kwargs):
+        """ Ensure the class has essential tags """
+        super()._setup(*args, **kwargs)
+        self._completed = None
+        self._na = None
+        self._no = None
+
+    def readline(self, *args, **kwargs):
+        line = super().readline(*args, **kwargs)
+        if self._completed is None and "ORCA TERMINATED NORMALLY" in line:
+            self._completed = True
+        elif self._na is None and "Number of atoms" in line:
+            v = line.split()
+            self._na = int(v[-1])
+        elif self._no is None and "Number of basis functions" in line:
+            v = line.split()
+            self._no = int(v[-1])
+        return line
+
+    readline.__doc__ = SileORCA.readline.__doc__
+
+    @sile_fh_open()
+    def completed(self):
+        """ True if the full file has been read and "ORCA TERMINATED NORMALLY" was found. """
+        if self._completed is None:
+            completed = self.step_to("ORCA TERMINATED NORMALLY")[0]
+        else:
+            completed = self._completed
+        if completed:
+            self._completed = True
+        return completed
+
+    @property
+    @sile_fh_open()
+    def na(self):
+        """ Number of atoms """
+        if self._na is None:
+            f = self.step_to("Number of atoms")
+            if f[0]:
+                self._na = int(f[1].split()[-1])
+            else:
+                return None
+        return self._na
+
+    @property
+    @sile_fh_open()
+    def no(self):
+        """ Number of orbitals (basis functions) """
+        if self._no is None:
+            f = self.step_to("Number of atoms")
+            if f[0]:
+                self._no = int(f[1].split()[-1])
+            else:
+                return None
+        return self._no
+
+    @sile_fh_open()
```

Comment:
yes, so you force the reopen on the first step_to, instead use `sile_fh_open(True)`, clearer, and removing the `reopen` argument clarifies intent

Reply:
I believe this has been addressed with the latest commits